client/src/components/Sharing/Sharing.vue

```diff
@@ -0,0 +1,469 @@
+                this.setSharing(this.actions.disable_link_access);
+            }
+        },
+        onPublish(published) {
+            if (published) {
+                this.item.importable = true;
+                this.setSharing(this.actions.publish);
+            } else {
+                this.setSharing(this.actions.unpublish);
+            }
+        },
+        getModel() {
+            this.ready = false;
+            axios
+                .get(`${getAppRoot()}api/${this.pluralNameLower}/${this.id}/sharing`)
+                .then((response) => this.assignItem(response.data))
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        setUsername() {
+            const Galaxy = getGalaxyInstance();
+            axios
+                .put(`${getAppRoot()}api/users/${Galaxy.user.id}/information/inputs`, {
+                    username: this.newUsername || "",
+                })
+                .then((response) => {
+                    this.hasUsername = true;
+                    this.getModel();
+                })
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        setSharing(action, user_id, share_option) {
+            let user_ids = undefined;
+            if (user_id) {
+                if (user_id.includes(",")) {
+                    user_ids = user_id.replace(/ /g, "").split(",");
+                } else user_ids = [user_id];
+            }
+
+            if (
+                action === this.actions.share_with &&
+                user_ids &&
+                this.item.users_shared_with.some(({ email }) => user_ids.includes(email))
+            ) {
+                this.addError(
+                    `You already shared this ${this.model_class} with
+                    ${user_ids.filter((user) => this.item.users_shared_with.some(({ email }) => email === user))}`
+                );
+                return;
+            }
+
+            const data = {
+                user_ids: user_ids,
+                share_option: share_option ? share_option : undefined,
+            };
+            return axios
+                .put(`${getAppRoot()}api/${this.pluralNameLower}/${this.id}/${action}`, data)
+                .then(({ data }) => {
+                    this.errors = [];
+                    this.assignItem(data);
+                    if (data.extra && data.extra.can_share) this.shareWithEmail = "";
+                    else this.shareWithEmail = user_id || "";
+                })
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        searchChanged(searchValue) {
+            if (searchValue === "") {
+                this.usersList = [];
+            } else {
+                axios
+                    .get(`${getAppRoot()}api/users?f_email=${searchValue}`)
+                    .then((response) => {
+                        this.usersList = response.data;
```

Comment:
You might want to restrict this to a minimum search value and possibly limit this on the backend. Returning all accounts containing `a` on the large instances is probably not what we'd want to do. 

We also probably want to make this conditional on `isExposeEmail`, if I am looking at this right the multiselect is not within `<div v-if="!isExposeEmail">`, so we should at least skip making the request here.

Reply:
> You might want to restrict this to a minimum search value and possibly limit this on the backend. Returning all accounts containing a on the large instances is probably not what we'd want to do.

done

> We also probably want to make this conditional on isExposeEmail, if I am looking at this right the multiselect is not within <div v-if="!isExposeEmail">, so we should at least skip making the request here.

it's conditional on `isExposeEmail`